internal/database/service.go

```diff
@@ -59,6 +59,47 @@ func (s *Neo4jService) ExecuteWriteQuery(ctx context.Context, cypher string, par
 	return res.Records, nil
 }
 
+// GetQueryType prefixes the provided query with EXPLAIN and returns the query type (e.g. 'r' for read, 'w' for write, 'rw' etc.)
+// This allows read-only tools to determine if a query is safe to run in read-only context.
+func (s *Neo4jService) GetQueryType(ctx context.Context, cypher string, params map[string]any, database string) (string, error) {
+	if s.driver == nil {
+		err := fmt.Errorf("neo4j driver is not initialized")
+		log.Printf("error in GetQueryType: %v", err)
+		return "", err
+	}
+
+	explainedQuery := "EXPLAIN " + cypher
+	res, err := neo4j.ExecuteQuery(ctx, *s.driver, explainedQuery, params, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase(database))
```

Comment:
Does this double the read query load? Do you think we should document that as well? Caching the query type results for identical queries could be a long-term solution but I don't think we are there yet:)

Reply:
I agree on the cache mechanism for database invocations, I think it will be out of the scope for the project for long time or ever.